controller/replica_controller.go

```diff
@@ -215,6 +230,17 @@ func (rc *ReplicaController) syncReplica(key string) (err error) {
 	}
 
 	if replica.DeletionTimestamp != nil {
+		if orphaned, err := rc.isOrphaned(replica); err != nil {
+			return err
+		} else if orphaned {
+			// replica adoption
+			replica.Spec.DataPath = ""
```

Comment:
Add an event log and a comment here, since we won't able to clean up the leftover of the replica.

And now multiple controllers are racing to get the ownership of the replica, which will cause conflict. But that seems hard to avoid.

Reply:
Will do.

I realized, but I couldn't think of a better way besides electing a leading manager to handle orphaned replicas. The race effectively elects a leader, but it's still a race.

Reply:
Maybe we can gather the similar situations in the controllers then use Kubernetes leader selection mechanism. One other place often caused conflict is the node status update part.

Reply:
Good idea - let's keep it in memory